linux/hid.c

```diff
@@ -582,6 +754,53 @@ struct hid_device_info  HID_API_EXPORT *hid_enumerate(unsigned short vendor_id,
 					 * check for USB and Bluetooth devices above */
 					break;
 			}
+
+			/* Construct <sysfs_path>/device/report_descriptor */
+			char *rpt_path = (char *)calloc(1, strlen(sysfs_path) + 25);
+			sprintf(rpt_path, "%s/device/report_descriptor", sysfs_path);
```

Comment:
1) Correct me if I'm wrong, but 25 is not enough to hold a NULL-terminator
2) we probably want to use `snprintf` instead